### ISSUE TITLE
Missing Avatar fetch after update.

### ISSUE THREAD

**Issue by xXSTrikeXx** [nextcloud/android#2196]
Hey,

after updating to v3.0.0 my profile avatar isnt fetching from the server. I think its only after upgrading to the new version and not for new app registrations. Im using Huawei Y6 2017.

**Comment by tobiasKaminsky** [comment]
Which server version do you use?
Do I understand you correctly that with an upgraded app (2.x -> 3.0) it is working, whereas a freshly installed app does not get the avatar?

**Comment by xXSTrikeXx** [comment]
No. Exatcly inversely. I updated from v2.X android to 3.0.X and server to v13. And after that my profile account hasnt got any image. On friends mobile phone (Android 4, Galaxy S3 Neo) it is working, but he DOESN´T use an own avatar. There is only the standard avaratar with the first letter of his name. I tested new accounts adding to the app and it is working on my device (but I tested it with not an own avatar. Just first letter of name).

**Comment by tobiasKaminsky** [comment]
@xXSTrikeXx on the drawer, what is the name before the "@"?
Does this match the username in user list when logged in as admin?

**Comment by xXSTrikeXx** [comment]
The name before "@" is the same as the login name. It´s my name "Timo".

If I understand your 2nd question correctly its the same name as the name when editing users in the web ;)

**Comment by tobiasKaminsky** [comment]
Strange...
The default avatar (Letter with color) can also be generated on device as a fallback, so this does not mean anything.
Can you create us a test account, test if the problem occurs also there and if so send the credentials to android at nextcloud dot com with a reference to this issue?

**Comment by xXSTrikeXx** [comment]
Done. 👍

**Comment by xXSTrikeXx** [comment]
![screenshot_2018-02-27-20-36-14](https://user-images.githubusercontent.com/1523741/36750562-20d9b0c4-1bfe-11e8-8221-10caddc86412.png)

Updating Libraries (3.0.2) don't resolve the problem. I'm the only one who have this behavior? When this is the case I will reinstall the app and close the issue ;)

**Comment by AndyScherzinger** [comment]
Never had the issue strangely you don't see any avatar,not even a fallback. That is extremely strange...

**Comment by tobiasKaminsky** [comment]
@xXSTrikeXx thanks for your email with test credentials.
For me it worked without a problem:
- first letter T on green background
- changing avatar on web ui
- waiting a bit
- new avatar on drawer

As you said in your email, I'll provide you some test credentials on my server.

**Comment by xXSTrikeXx** [comment]
@tobiasKaminsky  I deinstalled the app and reinstalled. Do I have to delete all old files before reinstalling it? Because I become the same behavior after a while after reinstalling the app.

**Comment by xraMsamohT** [comment]
I can confirm that an upgraded app does not load any custom avatar. Neither in the head of the sidebar nor in the account settings of the app. Clearing the app cache does not solve the issue.

**Comment by xXSTrikeXx** [comment]
@xraMsamohT Thanks for confirming this bug. I already reinstalled without clearing the cache. 

FYI on Talk the Avatar is displayed correct!

**Comment by xraMsamohT** [comment]
@xXSTrikeXx Galaxy Note 4 with stock Android 6.0.1.

Nextcloud 12.0.5
Android app 2.x -> 3.0.0 ... 3.0.3

**Comment by xXSTrikeXx** [comment]
It seeems, that it is an Android 6.x error. Any ideas? 
@tobiasKaminsky @AndyScherzinger @mario :)

**Comment by xraMsamohT** [comment]
I reanimated my old Nexus S with CyanogenMod 11 (Android 4.4.4). No problem with the avatars. 😉

**Comment by xXSTrikeXx** [comment]
@xraMsamohT year, I don't have any problems on new/other devices. AVD shows me and all works fine. It has to do with the updates ;)

**Comment by tobiasKaminsky** [comment]
@xXSTrikeXx can you try the failing app with a new account? 
I am unsure if we talked about this, but I can create you a test account on a different server.
Can you provide us additional infos via logcat? https://github.com/nextcloud/android/blob/master/README.md#getting-debug-info-via-logcat

**Comment by xXSTrikeXx** [comment]
@tobiasKaminsky As you can see in my last screenshot I´ve already added a new account to my failing app.
And yes we´ve already talked about this, and you have already send me account credentials to your E2E server 😉 . I think I´ve answered directly to your email 😃 .
Search for:
03-08 15:15:46.522 20042 20503 D OwnCloudClient #9: Creating OwnCloudClient
There i Readded my test account ;)
Please have a look on line 19283:
03-08 15:16:50.541 20898 20927 D Avatar  : URI: https://....

I added the test account I´ve created for you and created a new logcat. 2nd link ;) I sent it on IRC in pm to you ;)

**Comment by xXSTrikeXx** [comment]
Latest update rc1 doesn't solve it

**Comment by xXSTrikeXx** [comment]
Even 3.1 let me stay on this old problem. Any solution? Maybe a complete reinstall? How can I purge the database? Because I think this is the reason.

**Comment by xraMsamohT** [comment]
Same here. A reinstall with a sync of dozens of gigabytes is not a solution for me. ;-)

**Comment by xraMsamohT** [comment]
Reinstall (uninstall and install) removes all synced files on the smartphone, imho. That's dozens of gigabytes. Reloading all files after upgrade to v3.0 was a pain.

**Comment by tobiasKaminsky** [comment]
Re-installing the app would lead to remove of all downloaded files.
If you really want, you can move the downloaded folder, then re-install the app and move it back in again.
I am unfortunately a bit out of clue, what the problem could be. Therefore I also cannot try to debug it with you, as I would at least need one single clue...

**Comment by xraMsamohT** [comment]
> If you really want, you can move the downloaded folder, then re-install the app and move it back in again.

That hasn't work when upgrading to v3.0. That's why I do not want to repeat that for testing.

**Comment by xXSTrikeXx** [comment]
V.3.2 RC2 seems to fix this, can others approve, that the bug is fixed in this version?

**Comment by AndyScherzinger** [comment]
Well, I get the same behavior/issue again with 3.3.0 RCs :/, see #3015

**Comment by tobiasKaminsky** [comment]
Is this still the case? I recall that we fixed this. 
It was due to a changed response on eTag usage?
